terraform-environments/hosts-user-data.sh

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/bash
+
+
+# set variables
+  sudo_path=/bin/sudo
+  apt_path=/bin/apt
+  curl_path=/bin/curl
+  sed_path=/bin/sed
+
+# USERNAME=$username
+set -u
+set -e
+set +x
+set -o pipefail
+
+cd /
+
+# install python3.11
+$sudo_path $sed_path -i "s/#\$nrconf{kernelhints} = -1;/\$nrconf{kernelhints} = -1;/g" /etc/needrestart/needrestart.conf
+$sudo_path $sed_path -i 's/#$nrconf{restart} = '"'"'i'"'"';/$nrconf{restart} = '"'"'a'"'"';/g' /etc/needrestart/needrestart.conf
+$sudo_path $apt_path update && $sudo_path $apt_path -y upgrade
+$sudo_path $apt_path install python3.11 -y
+
+$sudo_path $apt_path install ssh -y
+
+# #install and start up venv
+# $sudo_path $apt_path install python3.11-venv -y
+# $sudo_path mkdir switcheroo_venv
+# cd switcheroo_venv
+# $sudo_path python3.11 -m venv .venv
+# source .venv/bin/activate
+
+# #install pip into venv - don't need its already there
+$sudo_path $apt_path install python3.11-dev python3.11-distutils -y
+$sudo_path $curl_path -O https://bootstrap.pypa.io/get-pip.py
+$sudo_path python3.11 get-pip.py
+
+
+
+# test installation of pylint to ensure pip is installed 
+$sudo_path pip3.11 install pylint
```

Comment:
I would avoid installing a random package because its dependencies might mess with ours. Probably best to just check the output of pip --version or something